File: 45secondapps_ui/src/services/ApplicationDeveloperService.js
import http from "../http-common"; 

class ApplicationDeveloperService {
  getAllApplicationDevelopers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/applicationDeveloper/applicationDevelopers`, searchDTO);
  }

  get(applicationDeveloperId) {
    return this.getRequest(`/applicationDeveloper/${applicationDeveloperId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/applicationDeveloper?field=${matchData}`, null);
  }

  addApplicationDeveloper(data) {
    return http.post("/applicationDeveloper/addApplicationDeveloper", data);
  }

  update(data) {
  	return http.post("/applicationDeveloper/updateApplicationDeveloper", data);
  }
  
  uploadImage(data,applicationDeveloperId) {
  	return http.postForm("/applicationDeveloper/uploadImage/"+applicationDeveloperId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ApplicationDeveloperService();

File: src/main/java/com/45secondapps/service/ApplicationDeveloperService.java
package com.45secondapps.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.45secondapps.domain.ApplicationDeveloper;
import com.45secondapps.dto.ApplicationDeveloperDTO;
import com.45secondapps.dto.ApplicationDeveloperSearchDTO;
import com.45secondapps.dto.ApplicationDeveloperPageDTO;
import com.45secondapps.dto.ApplicationDeveloperConvertCriteriaDTO;
import com.45secondapps.service.GenericService;
import com.45secondapps.dto.common.RequestDTO;
import com.45secondapps.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ApplicationDeveloperService extends GenericService<ApplicationDeveloper, Integer> {

	List<ApplicationDeveloper> findAll();

	ResultDTO addApplicationDeveloper(ApplicationDeveloperDTO applicationDeveloperDTO, RequestDTO requestDTO);

	ResultDTO updateApplicationDeveloper(ApplicationDeveloperDTO applicationDeveloperDTO, RequestDTO requestDTO);

    Page<ApplicationDeveloper> getAllApplicationDevelopers(Pageable pageable);

    Page<ApplicationDeveloper> getAllApplicationDevelopers(Specification<ApplicationDeveloper> spec, Pageable pageable);

	ResponseEntity<ApplicationDeveloperPageDTO> getApplicationDevelopers(ApplicationDeveloperSearchDTO applicationDeveloperSearchDTO);
	
	List<ApplicationDeveloperDTO> convertApplicationDevelopersToApplicationDeveloperDTOs(List<ApplicationDeveloper> applicationDevelopers, ApplicationDeveloperConvertCriteriaDTO convertCriteria);

	ApplicationDeveloperDTO getApplicationDeveloperDTOById(Integer applicationDeveloperId);







}





